tests/test_cms_menus.py

```diff
@@ -250,6 +253,115 @@ def test_get_roots_with_versioning_disabled(self):
             list(roots), [menucontent_1.root, menucontent_2.root, menucontent_3.root]
         )
 
+    def test_draft_menu_on_draft_page(self):
+        """
+        Draft menu changes on draft page rendering.
+        """
+        menu_cont_draft = factories.MenuContentWithVersionFactory(version__state=DRAFT, language=self.language)
+        menu_cont_published = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        pagecontent_aaa = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            title="aaa",
+            menu_title="aaa",
+            page_title="aaa",
+            version__state=DRAFT,
+        )
+        pagecontent_bbb = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            title="bbb",
+            menu_title="bbb",
+            page_title="bbb",
+            version__state=PUBLISHED,
+        )
+
+        child = factories.ChildMenuItemFactory(parent=menu_cont_draft.root, content=pagecontent_aaa.page)
+        child_1 = factories.ChildMenuItemFactory(parent=menu_cont_published.root, content=pagecontent_bbb.page)
+
+        # Node added in draft menu version is rendered in page draft view only
+        draft_page_endpoint = get_object_edit_url(pagecontent_aaa)
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(draft_page_endpoint)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(child.title, str(response.content))
+        self.assertNotIn(child_1.title, str(response.content))
+
+    def test_draft_menu_on_published_page(self):
+        """
+         Draft menu changes on published page
```

Comment:
The comment is basically the same as the test name, add value or don't add it at all.

```suggestion
         Ensure that a published page only shows a published menu
```